### ISSUE TITLE
about WIFIKILL. 

### ISSUE THREAD

**Issue by reigelgallarde** [pihomeserver/Kupiki-Hotspot-Script#150]
This is not an issue about the script but can be an enhancement.

Has anyone here know about wifikill? https://forum.xda-developers.com/showthread.php?t=1282900

There are also other apps that does the same.
Surprisingly for me, that it  can disable internet connection for a device on the same network.

What it do is it scans the network and do its thing.
![image](https://user-images.githubusercontent.com/10612336/49623878-c702b980-fa12-11e8-809e-bcd9b0266427.png)

I'm trying to find a solution.
One thing I could think of is if this app is not able to see who's on the network, this app can not do its thing.

Maybe isolate the network using iptables?
So any device will not be able to check on the network.
The same as what we have now `"Block access from LAN to WAN except portal"`
But I'm not good at iptables, so I can't do a PR for this.

**Comment by pihomeserver** [comment]
Looking at the comments from the developper it seems that the device use ARP reply requests to act as a router and then it drops packets of the selected device to keep it on the wifi network but with no internet access
That's not something that could be done on the router side from my understanding. But it seems that you can isolate each device from the others using PVLAN but then i don't know how it works in combination with Coova 
Interesting point :)

**Comment by reigelgallarde** [comment]
Yeah, I've read an interesting [post here](https://security.stackexchange.com/a/90244) . Answering **How to be safe from wifikill!**

> Some access points have a special mode of operation - sometimes called "Private Mode", "Client Isolation", "Guest Mode" or "Privacy" - whereby, however large the pool of devices connected to the AP, you only see yours. You cannot "talk" to the other devices. The AP will accept packets from you to it, and not forward any other except for NATting you on the Internet.

I'm still looking for possible ways but thinking if there's only a way to create or mimic PVLAN using software to isolate the clients, that would be great.

I've tried `ap_isolate` of hostapd and it works for some apps but not on wifikill.
what's bothering me most is that wifikill can look up mac addresses on the network.
Which is a sensitive information that is needed to be hidden to prevent mac spoofing.

**Comment by pihomeserver** [comment]
If you know how to test as i do not have an Android phone you can try to run this commands : 
Get the Mac address of your eth0 with 
```
ifconfig eth0
```
Then run 
```
arptables -P INPUT DROP
arptables -A INPUT --source-mac AA:BB:CC:DD:EE -j ACCEPT
```
